docs/index.md

```diff
@@ -251,6 +248,131 @@ $ git clone https://github.com/grpc/grpc.git
+
+To install protocol buffers support (the protocol buffer compiler and the protocol buffers runtime) with gRPC for your chosen language:
+
+<div class="tabs">
+  <ul>
+    <li><a href="#java_protoc">Java</a></li>
+    <li><a href="#cpp_protoc">C++</a></li>
+    <li><a href="#python_protoc">Python</a></li>
+    <li><a href="#go_protoc">Go</a></li>
+    <li><a href="#ruby_protoc">Ruby</a></li>
+    <li><a href="#node_protoc">Node.js</a></li>
+    <li><a href="#csharp_protoc">C#</a></li>
+    <li><a href="#objective-c_protoc">Objective-C</a></li>
+    <li><a href="#php_protoc">PHP</a></li>
+  </ul>
+<div id="java_protoc">
+<p>Again, the example is also part of Java gRPC itself's build, and includes pre-generated gRPC code - we recommend that you use this when working through this example.
+
+<p>See the <a href="https://github.com/grpc/grpc-java/blob/master/README.md">README</a> for instructions on how to install the protocol buffer compiler for your own projects.
+</div>
+<div id="cpp_protoc">
+If you <a href="https://github.com/grpc/grpc/blob/{{ site.data.config.grpc_release_branch }}/INSTALL.md">build and install gRPC from source</a>, the Makefile will automatically install <code>protoc</code> as well (if you don't already have it installed). Installing gRPC also gets you the gRPC C++ plugin for <code>protoc</code>.
+</div>
+<div id="python_protoc">
+First <a href="https://github.com/google/protobuf/blob/master/README.md">install <code>protoc</code></a> from its Github repository.
+
+<!--Where do they get the gRPC protoc plugin? Also what's the command to install the protobuf wheel, and do they need that too?-->
+</div>
+<div id="go_protoc">
+First <a href="https://github.com/google/protobuf/blob/master/README.md">install <code>protoc</code></a> from its Github repository. Then get the Go <code>protoc</code> plugin:
+
+<pre class="prettyprint">
+$ go get -a github.com/golang/protobuf/protoc-gen-go
+</pre>
+</div>
+<div id="ruby_protoc">
+First <a href="https://github.com/google/protobuf/blob/master/README.md">install <code>protoc</code></a> from its Github repository.
+
+<!--Where do they get the gRPC protoc plugin? Also what's the command to install the protobuf gem, and do they need that too?-->
+</div>
+<div id="node_protoc">
+The Node.js gRPC library currently dynamically generates the necessary gRPC code at runtime, so you don't need to install a protocol buffer compiler yourself.
+</div>
+<div id="csharp_protoc">
+On Mac, Linux, and Windows, install the <a href="https://www.nuget.org/packages/Google.Protobuf">Google.Protobuf</a> and <a href="https://www.nuget.org/packages/Grpc.Tools">gRPC.Tools</a> NuGet packages from your IDE (Visual Studio, Monodevelop, Xamin Studio).
```

Comment:
for C# we only ship precompiled   protoc and grpc_csharp_plugin  for Windows platform.
For Linux and Mac, protoc and grpc_csharp_plugin still need to be installed using Homebrew.  

Also, this is changing soon and we will be shipping precompiled  protoc and grpc_csharp_plugin for all of Windows, Mac and Linux (possibly in the next release).